src/main/java/com/fortify/plugin/jenkins/steps/FortifyUpload.java

```diff
@@ -301,14 +324,22 @@ public Long runWith(FortifyClient client) throws Exception {
 		}
 	}
 
-	private void pollFprProcessing(Long artifactId, TaskListener listener) throws IOException {
+	private void pollFprProcessing(Run<?, ?> run, Long artifactId, TaskListener listener) throws IOException {
 		PrintStream log = listener.getLogger();
 		boolean isProcessingComplete = false;
+
+		final long startTimeMillis = System.currentTimeMillis();
+
+		int sleepInMinutes = (getResolvedPollingInterval(listener) != null) ? getResolvedPollingInterval(listener) : 1;
+		int sleepInMillis = sleepInMinutes * 60 * 1000;
+		long sleepUntil = startTimeMillis + sleepInMillis;
+
+		int timeoutInMinutes = (getResolvedTimeout(listener) != null) ? getResolvedTimeout(listener) : 0;
+		int timeoutInMillis = timeoutInMinutes * 60 * 1000;
+		long timeoutAfter = startTimeMillis + timeoutInMillis;
+
 		while (!isProcessingComplete) {
-			int sleep = (getResolvedPollingInterval(listener) != null) ? getResolvedPollingInterval(listener) : 1;
-			log.printf("Sleep for %d minute(s)%n", sleep);
-			sleep = sleep * 60 * 1000; // wait time is in minute(s)
-			long sleepUntil = System.currentTimeMillis() + sleep;
+			log.printf("Sleep for %d minute(s)%n", sleepInMinutes);
 			while (true) {
 				long diff = sleepUntil - System.currentTimeMillis();
```

Comment:
Extraction of sleepUntil variable outside of while(!isProcessingComplete) breaks the thread sleeping in polling logic. If the idea is to poll server with some pauses, in your case the while cycle with sleeping code will be executed only ones and after that it will be skipped and poll request would be invoked without any pauses. Because sleepUntil is not recalculated anymore for each cycle.

Reply:
Looks like you are right. I have rolled back this change.